website/app.py

```diff
@@ -33,19 +46,21 @@ def location_page():
     return render_template('location_form.html')
 
 
-@app.route('/submit-location', methods=['POST'])
+@app.route('/submit-user', methods=['POST'])
 def submit_location():
-    """When users submit a location, the information gets directed here, where it
-    get parsed and uploaded."""
-    df = get_df()
-    location_type = request.form['locationType']
-    location_value = request.form[location_type]
-    if location_type == 'postcode':
-        longitude, latitude = get_long_lat(location_value.upper(), df)
-        location_name = get_location_name(location_value.upper(), df)
-        conn = get_db_connection(ENV)
-        add_to_databse('location', (location_name, longitude, latitude), conn)
-    return f"Location submitted: {location_type} - {location_value}"
+    """When a user signs-up, it gets sent through here and uploaded
+       to the database."""
+    location_value = request.form['location']
+    name = request.form['name']
+    email = request.form['email']
+    sub_newsletter = request.form.get('newsletter', 'off') == 'on'
+    sub_alerts = request.form.get('alerts', 'off') == 'on'
+    details = get_details_from_post_code(location_value)
+    conn = get_db_connection()
+    if details['status'] == 200:
+        setup_user_location(details, name, email,
+                            sub_newsletter, sub_alerts, conn)
+    return render_template('cant_be_found_page.html')
```

Comment:
does this always return a cant_be_found_page? should this not be in an "else", with status 200 returning something else?

Reply:
fixed